src/snowflake/cli/_plugins/nativeapp/entities/application_package.py

```diff
@@ -556,10 +650,24 @@ def _bundle(self):
         )
         compiler = NativeAppCompiler(bundle_context)
         compiler.compile_artifacts()
+
+        if self._entity_model.children:
+            # Append children SQL to setup script
+            setup_file_path = find_setup_script_file(deploy_root=self.deploy_root)
+            with open(setup_file_path, "r", encoding="utf-8") as file:
```

Comment:
let's think of ways to unify this with the processors in the future. Maybe this is just a processor? Maybe we need to extract some logic?

Reply:
It's too soon for such a refactor in this PR, but it's a good idea to consider once this feature stabilizes. I'll add a comment here for now.